controller_interface/include/controller_interface/chainable_controller_interface.hpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2022, Stogl Robotics Consulting UG (haftungsbeschränkt)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CONTROLLER_INTERFACE__CHAINABLE_CONTROLLER_INTERFACE_HPP_
+#define CONTROLLER_INTERFACE__CHAINABLE_CONTROLLER_INTERFACE_HPP_
+
+#include <vector>
+
+#include "controller_interface/controller_interface.hpp"
+#include "controller_interface/visibility_control.h"
+#include "hardware_interface/handle.hpp"
+
+namespace controller_interface
+{
+/// Virtual class to implement when integrating a controller that can be preceded by another
+/// controllers.
+/**
+ * Specialization of ControllerInterface class to force implementation of methods specific for
+ * "chainable" controller, i.e., controller that can be preceded by an another controller, for
+ * example inner controller of an control cascade.
+ *
+ */
+class ChainableControllerInterface : public ControllerInterface
+{
+public:
+  CONTROLLER_INTERFACE_PUBLIC
+  ChainableControllerInterface() = default;
+
+  CONTROLLER_INTERFACE_PUBLIC
+  virtual ~ChainableControllerInterface() = default;
+
+  CONTROLLER_INTERFACE_PUBLIC
+  return_type update(const rclcpp::Time & time, const rclcpp::Duration & period) final;
+
+  CONTROLLER_INTERFACE_PUBLIC
+  bool is_chainable() const final;
```

Comment:
Why `final` on all these? What if people want to override these implementations in a derived class?

Let's say, you want to derive a controller from JTC (of which there were a few examples in ROS1)

Reply:
Hmmmmmm, maybe you are right... this means you would be able to make inherited controller chainable even the parent controller is not...

I was thinking more about very strict separation between "normal" and chainable controllers. There is some chainable-controllers specific management in the base class which should be then done manually. So this could end up very bad in many cases...